### ISSUE TITLE
"ValueError: missing port number in address '--no-dashboard'"

### ISSUE THREAD

**Issue by arokem** [pangeo-data/pangeo#715]
Hello everyone! 

After an update of my pangeo with a new docker image (adding a Python dependency), I am running into this error now when trying to start a cluster. Does anyone know how I might debug/resolve this? 

Thanks!

**Comment by jhamman** [comment]
@arokem - can you report on the version numbers for dask/distributed/dask-kubernetes?

**Comment by arokem** [comment]
dask is at 2.1.0, distributed at 1.28.1 and dask_kubernetes is at 0.9.0

**Comment by arokem** [comment]
While addressing #717, distributed seems to have jumped to 2.1.0, resolving this issue along the way. 

So, question: I have a custom docker configuration that starts with: 

    from pangeo/base-notebook

And then goes on to pip and conda install a few other specific dependencies. Should I pin the version of dak, distributed and dask_kubernetes in the requirements file used as input to pip?

**Comment by jhamman** [comment]
It should be safe to differ to base-notebook to give you versions of dask/distributed/kubernetes that work together. You probably don't need to install them at all. Are you using the `-onbuild` variants of the docker images in pangeo-stacks?

**Comment by arokem** [comment]
I'm not 100% sure, but I think not. 

Here is my docker file: 

```
from pangeo/base-notebook

USER root
RUN apt-get update \
    && apt-get install -yq --no-install-recommends less sudo octave graphviz nano\
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/*
RUN pip install --upgrade pip
RUN pip install octave_kernel && python -m octave_kernel install
ADD requirements.txt .
RUN pip install -r requirements.txt
RUN conda install --name notebook -y -c pennmem ptsa==2.0.4
RUN jupyter labextension install @pyviz/jupyterlab_pyviz
RUN jupyter labextension install @lckr/jupyterlab_variableinspector
USER jovyan
```

And requirements.txt: 

```
h5py==2.9.0
matplotlib==3.1.0
nitime==0.7
xarray==0.12.1
numba==0.43.1
scikit-image==0.15.0
gcsfs==0.2.2
oct2py==5.0.2
zarr==2.3.1
fooof==0.1.3
neurodsp==1.1.2
bycycle==0.1.2
pactools
seaborn==0.9.0
holoviews==1.12.3
datashader==0.7.0
hvplot==0.4.0
panel==0.5.1
tqdm==4.32.1
plotly==3.9.0
ipywidgets==7.4.2
intake==0.5.1
prefect==0.6.3
mne==0.18.2
jupyterlab==1.1.2
```

Should I add something like [this](https://github.com/pangeo-data/pangeo-stacks/blob/master/onbuild/Dockerfile) into my dockerfile?

**Comment by jhamman** [comment]
I think your dockerfile could basically look like this:

```dockerfile
from pangeo/base-notebook-onbuild
```

Then you can use _normal_ repo2docker configs. For example, you can put all your conda/pip installs into `environment.yaml`:

```yaml
name: pangeo
channels:
  - pennmem 
  - conda-forge
dependencies:
  - ptsa==2.0.4
  - h5py==2.9.0
  - matplotlib==3.1.0
  - nitime==0.7
  - xarray==0.12.1
  - numba==0.43.1
  - scikit-image==0.15.0
  - gcsfs==0.2.2
  - oct2py==5.0.2
  - zarr==2.3.1
  - fooof==0.1.3
  - neurodsp==1.1.2
  - bycycle==0.1.2
  - pactools
  - seaborn==0.9.0
  - holoviews==1.12.3
  - datashader==0.7.0
  - hvplot==0.4.0
  - panel==0.5.1
  - tqdm==4.32.1
  - plotly==3.9.0
  - ipywidgets==7.4.2
  - intake==0.5.1
  - prefect==0.6.3
  - mne==0.18.2
  - jupyterlab==1.1.2
  - pip
  pip:
    - octave_kernel # you can also get this from conda-forge btw
    - other-things-from-requirements.txt
   ```

_Note that I've put all your pip installs in conda land. We're working on solidifying this as a best practice and if it works for you, I'd try to avoid using pip here_

Create an `apt.txt` file
```
less
sudo
octave
graphviz
nano
```

and a `postBuild` file for everything else:

```bash
jupyter labextension install @pyviz/jupyterlab_pyviz
jupyter labextension install @lckr/jupyterlab_variableinspector
```

**Comment by arokem** [comment]
OK - I have finally been able to try this out (actually on my [new AWS deployment](https://github.com/pangeo-data/pangeo-cloud-federation/issues/416)). I created a configuration as you suggested ([here](https://github.com/nrdg/hcp-pangeo-docker)), built and pushed it using jupyter-repo2docker and then pointed my deployment to use this image. The image seems to get pulled fine into the hub, but I can't import packages specified in the `environment.yaml` file. Should there be a conda env called `pangeo` visible in the hub? Currently, there is only one non-base env in there, called notebook. Is there some step missing?
